src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -311,6 +319,162 @@ describe('WorkspacesUsecases', () => {
     });
   });
 
+  describe('changeUserRole', () => {
+    it('When team does not exist, then error is throw', async () => {
+      jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(null);
+
+      await expect(
+        service.changeUserRole('workspaceId', 'teamId', {
+          userId: 'userId',
+          role: WorkspaceRole.MEMBER,
+        }),
+      ).rejects.toThrow(NotFoundException);
+    });
+
+    it('When user is not part of team, then error is throw', async () => {
+      jest
+        .spyOn(teamRepository, 'getTeamById')
+        .mockResolvedValue(newWorkspaceTeam());
+      jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(null);
+
+      await expect(
+        service.changeUserRole('workspaceId', 'teamId', {
+          userId: 'useriD',
+          role: WorkspaceRole.MEMBER,
+        }),
+      ).rejects.toThrow(NotFoundException);
+    });
+
+    it('When user does not exist, then error is throw', async () => {
+      jest
+        .spyOn(teamRepository, 'getTeamById')
+        .mockResolvedValue(newWorkspaceTeam());
+      jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(
+        new WorkspaceTeamUser({
+          id: '',
+          teamId: '',
+          memberId: '',
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        }),
+      );
+      jest.spyOn(userRepository, 'findByUuid').mockResolvedValue(null);
+
+      await expect(
+        service.changeUserRole('workspaceId', 'teamId', {
+          userId: 'userId',
+          role: WorkspaceRole.MEMBER,
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('When user is member is updated to manager, then team manager is updated correctly', async () => {
+      const member = newUser();
+      const team = newWorkspaceTeam();
+
+      jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+      jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(
+        new WorkspaceTeamUser({
+          id: '',
+          teamId: team.id,
+          memberId: member.uuid,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        }),
+      );
+      jest.spyOn(userRepository, 'findByUuid').mockResolvedValue(member);
+
+      await service.changeUserRole('workspaceId', 'teamId', {
+        userId: 'userId',
+        role: WorkspaceRole.MANAGER,
+      });
+
+      expect(teamRepository.updateById).toHaveBeenCalledWith(team.id, {
+        managerId: member.uuid,
+      });
+    });
+
+    it('When user is manager is updated to member, then owner is assigned as manager', async () => {
```

Comment:
"When a team manager's role is changed to member, then the owner is assigned as manager" could be better